utils/misc.ml

```diff
@@ -549,13 +549,21 @@ let ordinal_suffix n =
 (* Color support handling *)
 module Color = struct
   external isatty : out_channel -> bool = "caml_sys_isatty"
+  external is_ansi_capable : out_channel -> bool = "caml_is_ansi_capable"
+  external set_ansi_capable : out_channel -> bool -> bool = "caml_set_ansi_capable"
+
+  let enable_win32_vt oc =
+    if is_ansi_capable oc then true
+    else if set_ansi_capable oc true then
+      (at_exit (fun () -> ignore (set_ansi_capable oc false)); true)
+    else
+      false
 
   (* reasonable heuristic on whether colors should be enabled *)
-  let should_enable_color () =
+  let should_enable_color oc =
     let term = try Sys.getenv "TERM" with Not_found -> "" in
-    term <> "dumb"
-    && term <> ""
-    && isatty stderr
+    (term <> "dumb" && term <> "" && isatty oc) ||
+    (Sys.win32 && enable_win32_vt oc)
```

Comment:
Perhaps Windows users will have a say better say on this but why not have:
```
(term <> "dumb" && Sys.win32 && enable_win32_vt oc)
```

Reply:
I know, that's the reason why I didn't suggest `term <> ""`. The reasoning is as follows.

But then once we are in Windows vt mode, I expect the program to behave like it would do on unix when it interacts with a tty (a nice touch from Microsoft would have been to have set the process' `TERM` environment variable to a suitable value when the mode is enabled).

And these programs interacting with `tty`s disable ANSI output when the environment variable `TERM` is `dumb`. 

This gives a way for windows users to disable coloring the standard way you would do on Unix.

Reply:
I made this change in the last commit even if I am not 100% convinced it is a good idea (but don't have strong arguments against either).

Reply:
I'm not a windows user, I think they should chime in to say what they think.

Reply:
I'm not terribly convinced, to be honest. `TERM` is part of Posix; VT100 is not - there is the part where we work out if we want coloured output (cf. `NO_COLOR` etc. in #10560) and then the part where we work out how to make those colours happen. On Unix, having decided we want to display in colour, we follow its standards and inspect `TERM` (which may cause us to discover that the terminal _does not support colour_) but on Windows, having decided we want to display, we follow its standards and select between either using the Windows API (which we have never done, but opam has) or using by the same API to indicate that we'd like VT100 enabled.

I'm all for cross-platform, but this seems unnecessarily Unix-ish on Windows for me!

Reply:
I don't mind, you be the judge. 

But note *for use in a more general setting*: people are confusing ANSI escapes with colors, but they provide much more than that (not to mention that the `NO_COLOR` "standard" was [a bad idea to start with](https://news.ycombinator.com/item?id=17244313), aka solving the problem at the wrong level). 

It seems to me that you'd still want a user mecanism on Windows to be able to tell the program "please don't treat that as an ANSI enabled terminal", on Unix setting `TERM=dumb` works for that, what is is the equivalent on Windows ?

Reply:
Agreed on the ANSI escapes. It's not necessarily fully coherent, but having given a Windows program the handle for the console, it is (from a native Windows perspective) strange to me to expect to be able to say "please don't do anything with it". I'd expect that the Windows equivalent of `TERM=dumb` would rather be not connecting stdout to a console in the first place. The bit that feels strange is that that `TERM` is meant to control what is written to the console, but there's still no reason not to control the Windows console using its API (it's neither deprecated nor particularly recommended that all programs start using VT100), and it certainly feels strange to have "TERM=dumb" starting to mean "don't use these syscalls please".

Slightly more coherently, it definitely does feel incorrect that we unilaterally adopt the interpretation of `TERM` this way (i.e. doing this with a lack of a standard) 👨‍⚖️

(FWIW when I first wrote that emulation in opam in 2014 https://github.com/dra27/opam/commit/02e86b48a03538cfdf2969705d3ce8c0763d9abc the support was indeed unconditional; apparently when I rebased it in 2015 past #1882, which introduced the `TERM=dumb` check, I was not concentrating in https://github.com/dra27/opam/commit/91c4a90f2b5cf9dbc96c28ade9c1e3b07ebdc61f, although there were a lot of commits to rebase back in those days...)